### ISSUE TITLE
Make only-new-issues work with Merge Queues, not just Pull Requests

### ISSUE THREAD

**Issue by andrewedstrom** [golangci/golangci-lint-action#956]
### Welcome

- [X] Yes, I understand that the GitHub action repository is not the repository of golangci-lint itself.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.

### Your feature request related to a problem? Please describe.

`only-new-issues` is fantastic, but it currently only works when the GitHub event_name is `pull_request`. Please add support for the `merge_group` event name, so that this feature can work with merge queues as well.

[Merge queues](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) are a powerful feature of GitHub for teams that merge many pull requests per day. It lets you queue up multiple pull requests to be merged in a row. Adding a PR to the queue triggers a new run of the status checks, and once the required status checks pass, that PR is merged.

### Describe the solution you'd like.

We just need `only-new-issues` to apply when the GitHub event is `merge_group`. I think this should be a one-line change in `src/run.ts`

The current code says:
```typescript
const ctx = github.context
  if (ctx.eventName !== `pull_request`) {
    core.info(`Not fetching patch for showing only new issues because it's not a pull request context: event name is ${ctx.eventName}`)
    return ``
  }
```

We should change this to
```typescript
const ctx = github.context;
if (ctx.eventName !== `pull_request` && ctx.eventName !== `merge_group`) {
  core.info(`Not fetching patch for showing only new issues because it's not a pull request or merge group context: event name is ${ctx.eventName}`);
  return ``;
}
```

### Describe alternatives you've considered.

My main alternatives are
1. Fix all linter errors in my project at once (hard)
2. Disable the linter as a required check.

**Comment by andrewedstrom** [comment]
@Sortren 

1. We make a pull request
2. This action runs and checks if any of the lines changed in the pull request have linter issues
3. We add the pull request to a merge queue and the status checks run again (this is the step where we have the problems raised in this issue)
4. Once status checks pass, the PR is merged into main.

**Comment by strantalis** [comment]
Our org is running into a similar issue as well. I probably missed something that don't understand about merge groups but this is what I changed. It seems to work but not sure if thats exactly what's needed to support merge groups.

https://github.com/strantalis/golangci-lint-action/commit/f4e979f5b36068d69d340547f66c991dca8b70ef

**Comment by flimzy** [comment]
@strantalis Did/can you createa  PR with those changes, to hopefully get a full review?

**Comment by ldez** [comment]
I create a test repo: I can never trigger the `merge_group` event.

https://github.com/golangci/friendly-lamp/blob/main/.github/workflows/merge-group.yml

Can anyone provide a working example of `merge_group`/merge queue?

**Comment by ldez** [comment]
Currently, `only-new-issues` only works for PR (events `pull_request` and `pull_request_target`) because it uses the GitHub API to get the diff from a PR (and not because of the `if` on the event name).

The addition of `` ctx.eventName !== `merge_group` `` will not work because `merge_group` event doesn't provide information about a PR.
The PR number can be found by parsing a ref name but this is not the right way because the diff will come from PR information and not from the rebased/merged/squashed branch commits of the queue.

The support of `merge_group` with `only-new-issues` requires writing a dedicated implementation, completely different than the PR support.

Currently, as a workaround to use `merge_group`, you should:
- not use`only-new-issues` but `args`
- use 2 conditional steps for golangci-lint
- set `fetch-depth` to `0`.

```yml
name: golangci-lint
on:
  pull_request:
  merge_group:

jobs:
  golangci-lint:
    name: lint
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 0

      - uses: actions/setup-go@v5
        with:
          go-version: stable
      
      - name: golangci-lint (PR)
        if: ${{ github.event_name == 'pull_request' }}
        uses: golangci/golangci-lint-action@v5
        with:
          version: v1.57
          only-new-issues: true

      - name: golangci-lint (queue)
        if: ${{ github.event_name == 'merge_group' }}
        uses: golangci/golangci-lint-action@v5
        with:
          version: v1.57
          args: --new-from-rev=${{ github.event.merge_group.base_sha }}
```

**Comment by ldez** [comment]
FYI, since v5.1.0, the event `pull_request`, `pull_request_target`, `push`, and `merge_group` are supported.

Note: for `merge_group`, you still need to set `fetch-depth` to `0`.

**Comment by strantalis** [comment]
@ldez Workflows seem to be working. Just wanted to say thanks for the help.
